manifests/transport_https.pp

```diff
@@ -0,0 +1,8 @@
+# Allow APT to use repos served over HTTPS
+class apt::transport_https (
+  $ensure = 'present'
+) {
+  package { ['apt-transport-https', 'ca-certificates']:
```

Comment:
I actually really dislike `ensure_packages`. I think it's the wrong solution for an admittedly annoying problem, because it's parse-order dependent. So one day it might work, and through no fault of your own the next it might not because a parse-order thing has changed and boom it goes.